src/gc.c

```diff
@@ -802,11 +812,17 @@ void jl_gc_force_mark_old(jl_ptls_t ptls, jl_value_t *v) JL_NOTSAFEPOINT
         jl_gc_queue_root(v);
 }
 
-#define should_collect() (__unlikely(gc_num.allocd>0))
-
 static inline int maybe_collect(jl_ptls_t ptls)
 {
-    if (should_collect() || gc_debug_check_other()) {
+    int should_collect = 0;
+    if (ptls->gc_num.allocd >= 0) {
+        int64_t intvl = per_thread_counter_interval(ptls->tid, gc_num.interval);
+        size_t localbytes = ptls->gc_num.allocd + intvl;
+        ptls->gc_num.allocd = -intvl;
+        jl_atomic_fetch_add(&gc_num.allocd, localbytes);
```

Comment:
`jl_atomic_fetch_add` should return the resulting value which is the right thing to use for the comparison on the next line.

Reply:
I think that's `add_fetch`? But yes we should use it.